R/utilities.R

```diff
@@ -122,22 +122,30 @@ throwWarningAboutMergedCategories <- function(affected.variables, function.name)
 #' @noRd
 warnIfSummingMultipleStatistics <- function(x, function.name)
 {
-    statistics <- lapply(x, lookupStatistics)
-    statistics <- Filter(Negate(is.null), statistics)
-    if (length(x) == 1L && length(statistics) > 0)
+    throw.warning <- FALSE
+    if (length(x) == 1L && isQTable(x[[1L]]))
     {
-        statistics <- statistics[[1L]]
-        throw.warning <- length(statistics) > 1L
-    } else if (length(statistics) > 1L)
+        qtable <- x[[1L]]
+        all.stats <- possibleStatistics(qtable)
+        throw.warning <- length(all.stats) > 1L
+        statistics <- if (throw.warning) possibleStatistics(qtable) else NULL
+    }
+    else if (length(x) > 1L)
     {
-        throw.warning <- !Reduce(identical, statistics)
-        statistics <- Reduce(union, statistics)
-    } else
-        throw.warning <- FALSE
+        all.stats <- lapply(x, possibleStatistics)
+        throw.warning <- !Reduce(identical, all.stats)
+        statistics <- if (throw.warning) Reduce(union, all.stats) else NULL
+    }
     if (throw.warning)
         throwWarningAboutDifferentStatistics(statistics, function.name)
 }
 
+statisticsPresentInLastDim <- function(qtable)
+{
+    span.dims <- sum(vapply(attr(qtable, "span"), ncol, integer(1L)))
```

Comment:
This line is not used.

Reply:
It's used on the next line? The `<` was mistaken for a `<-`?

Reply:
Haha, oops, my bad.